datadog_checks_dev/datadog_checks/dev/tooling/commands/metadata.py

```diff
@@ -0,0 +1,322 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import csv
+from collections import defaultdict
+from io import open
+
+import click
+from six import PY2, iteritems
+
+from .utils import CONTEXT_SETTINGS, abort, echo_failure, echo_warning
+from ..utils import get_metadata_file, get_metric_sources, load_manifest
+
+REQUIRED_HEADERS = {
+    'metric_name',
+    'metric_type',
+    'description',
+    'orientation',
+    'integration',
+}
+
+OPTIONAL_HEADERS = {
+    'interval',
+    'unit_name',
+    'per_unit_name',
+    'short_name',
+}
+
+ALL_HEADERS = REQUIRED_HEADERS | OPTIONAL_HEADERS
+
+VALID_METRIC_TYPE = {
+    'count',
+    'counter',
+    'distribution',
+    'gauge',
+    'rate',
+}
+
+VALID_ORIENTATION = {
+    '0',
+    '1',
+    '-1'
+}
+
+# To easily derive these again in future, copy the contents of `integration/system/units_catalog.csv` then run:
+#
+# pyperclip.copy('\n'.join("    '{}',".format(line.split(',')[2]) for line in pyperclip.paste().splitlines()))
+VALID_UNIT_NAMES = {
```

Comment:
Are all these valid? @l0k0ms @ofek ? Shouldn't we submit a warning for deprecated one?

Reply:
I just wanted to be sure. Thanks for the detailed reply 😃